dotcom-rendering/src/components/ContainerOverrides.tsx

```diff
@@ -0,0 +1,53 @@
+import { css } from '@emotion/react';
+import { isString, isUndefined } from '@guardian/libs';
+import { decideContainerOverrides } from '../lib/decideContainerOverrides';
+import type { palette } from '../palette';
+import type { DCRContainerPalette } from '../types/front';
+// import { useConfig } from './ConfigContext';
+
+type Props = {
+	containerPalette?: DCRContainerPalette;
+	isDynamo: boolean;
+	children: React.ReactNode;
+};
+
+/** @see https://developer.mozilla.org/en-US/docs/Web/CSS/display-box#contents */
+const displayContents = css`
+	display: contents;
+`;
+
+type ColourName = Parameters<typeof palette>[0];
+
+/**
+ * Add CSS custom property overrides for palette colours in a given container
+ */
+export const ContainerOverrides = ({
+	containerPalette,
+	isDynamo,
+	children,
+}: Props) => {
+	if (isUndefined(containerPalette)) return <>{children}</>;
```

Comment:
I'm curious about this line, why not make container palette defined? Is it because we can't guarantee we have the container palette on first render?

Reply:
Because of composition and the fact that a container may or may not have a palette. Without handling this case internally, you would have to handle the case higher up the chain with a ternary and it would probably be more confusing.

```jsx
// containerPalette may be undefined
const Parent = ({ containerPalette }) =>
  (containerPalette ?
     <ContainerOverrides containerPalette={containerPalette}>
       <Child />
     </ContainerOverrides>
    : <Child />
  )

```

Reply:
makes sense, thanks max!